RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc

```diff
@@ -0,0 +1,529 @@
+// -*- C++ -*-
+//
+// Package:    ElectronProducers
+// Class:      LowPtGsfElectronSeedProducer
+//
+/**\class LowPtGsfElectronSeedProducer RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc
+ Description: EDProducer of ElectronSeed objects
+ Implementation:
+     <Notes on implementation>
+*/
+// Original Author:  Robert Bainbridge
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/ParticleFlowReco/interface/PFTrajectoryPoint.h"
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+#include "FastSimulation/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "FastSimulation/Particle/interface/RawParticle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/TkClonerImpl.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/TkTransientTrackingRecHitBuilder.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
+#include "TrackingTools/PatternTools/interface/TrajectorySmoother.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/TrackFitters/interface/TrajectoryFitter.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TMath.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+LowPtGsfElectronSeedProducer::LowPtGsfElectronSeedProducer( const edm::ParameterSet& conf, 
+							    const lowptgsfeleseed::HeavyObjectCache* ) :
+  kfTracks_{consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"))},
+  pfTracks_{consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("pfTracks"))},
```

Comment:
It would be better to consume `"tracks"` only if `usePfTracks_ == false` (and `"pfTracks"` if it is `true`).

Reply:
Done.